core/domain/exp_jobs_one_off.py

```diff
@@ -492,42 +491,37 @@ def map(item):
-        # Convert the yaml content to dictionary.
-        htmlValidation_dict = utils.dict_from_yaml(htmlValidation_yaml)
-
-        allowed_parent_list = htmlValidation_dict['allowed_parent_list']
-        allowed_tag_list = htmlValidation_dict['allowed_tag_list']
-
-        for state in exploration.states.itervalues():
-            # result is the list of all the html present in the state.
-            result = exp_services.find_all_values_for_key(
-                'html', state.to_dict())
-
-            for html_data in result:
-                html_data = html_data.encode('utf-8')
-                soup = BeautifulSoup(html_data, 'html.parser')
-
-                for tag in soup.findAll():
-                    # Checking for tags not allowed in RTE.
-                    if tag.name not in allowed_tag_list:
-                        if 'invalidTags' in err_dict:
-                            err_dict['invalidTags'] += [tag.name]
-                        else:
-                            err_dict['invalidTags'] = [tag.name]
-                    # Checking for parent-child relation that are not
-                    # allowed in RTE.
-                    parent = tag.parent.name
-                    if (tag.name in allowed_tag_list and parent not in
-                            allowed_parent_list[tag.name]):
-                        if tag.name in err_dict:
-                            err_dict[tag.name] += [parent]
-                        else:
-                            err_dict[tag.name] = [parent]
-
-        for key in err_dict.iterkeys():
+        allowed_parent_list = (
+            feconf.RTE_CONTENT_SPEC['RTE_TYPE_TEXTANGULAR']
+            ['ALLOWED_PARENT_LIST'])
+        allowed_tag_list = (
+            feconf.RTE_CONTENT_SPEC['RTE_TYPE_TEXTANGULAR']
+            ['ALLOWED_TAG_LIST'])
+
+        html_list = exploration.get_all_html_content_strings()
+
+        for html_data in html_list:
+            html_data = html_data.encode('utf-8')
+            soup = bs4.BeautifulSoup(html_data, 'html.parser')
+
+            for tag in soup.findAll():
+                # Checking for tags not allowed in RTE.
+                if tag.name not in allowed_tag_list:
+                    if 'invalidTags' in err_dict:
+                        err_dict['invalidTags'] += [tag.name]
+                    else:
+                        err_dict['invalidTags'] = [tag.name]
+                # Checking for parent-child relation that are not
+                # allowed in RTE.
+                parent = tag.parent.name
+                if (tag.name in allowed_tag_list) and (
+                        parent not in allowed_parent_list[tag.name]):
+                    if tag.name in err_dict:
+                        err_dict[tag.name] += [parent]
+                    else:
+                        err_dict[tag.name] = [parent]
+
+        for key in err_dict:
             err_dict[key] = list(set(err_dict[key]))
 
         yield('Errors', err_dict)
```

Comment:
Just a thought -- worth doing a list(set()) on the values for err_dict? That way you don't get lots of repetition of the same problematic html, which might make it easier to get a sense of how many distinct cases there are.

Reply:
Yes, it is and I have also updated the reduce function to yield a better output. Something like this:

![screenshot from 2018-05-30 02-40-44](https://user-images.githubusercontent.com/22434689/40685944-39e24dc2-63b3-11e8-8a53-e9b0bc0e9034.png)